micromagneticdata/drive.py

```diff
@@ -415,3 +416,25 @@ def slider(self, description="step", **kwargs):
         return ipywidgets.IntSlider(
             value=0, min=0, max=self.n - 1, step=1, description=description, **kwargs
         )
+
+    def to_xarray(self):
+        """Export ``micromagneticdata.Drive`` as ``xarray.Dataset``"""
+        if self.info["driver"] == "MinDriver":
+            darray = self[0].to_xarray()
+        else:
+            field_darrays = (self[i].to_xarray() for i in range(self.n))
+            if self.info["driver"] == "TimeDriver":
+                darray = xr.concat(field_darrays, dim="t").assign_coords(
+                    {"t": self.table.data["t"].to_numpy()}
+                )
+            else:
+                darray = xr.concat(field_darrays, dim="B_hysteresis").assign_coords(
+                    {
+                        f"B{i}": ("B_hysteresis", self.table.data[f"B{i}_hysteresis"])
+                        for i in "xyz"
+                    }
+                )
```

Comment:
It would be nice if we could check if two of the hysteresis components are always zero (e.g. By == 0 and Bz == 0) and then automatically use Bx instead of `B_hysteresis` as `dim` for the DataArray. But maybe that is too much "magic"?

Reply:
I was also hoping that `xr.DataArray.plot()` has this magic built in but it does not. If we use `B_hysteresis` instead of `Bx_hysteresis` we lose the information about: which component is non-zero?

Reply:
Let's discuss this question tomorrow (as well) - it is complicated ;-)

One aspect to consider: every time we introduce some magic that makes the data very convenient to analyse *interactively*, we make it harder to analyse the data systematically (from a programme, for example).

To address this dilemma, we could have a convenience function `simplify` or `humanise_labels`  or something like that which contains the magic, and can either be de-activated, or needs to be called directly before it does anything. 

I don't fully understand where the current labels come from, but I think it will be most efficient to discuss in person.